src/ai/backend/manager/config/shared.py

```diff
@@ -0,0 +1,423 @@
+from __future__ import annotations
+
+import enum
+from datetime import datetime
+from ipaddress import IPv4Network
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, Field, IPvAnyNetwork
+
+from .local import HostPortPair
+
+
+class SystemConfig(BaseModel):
+    timezone: str = Field(
+        default="UTC",
+        description="""
+        Timezone setting for the manager.
+        Uses pytz-compatible timezone names.
+        Affects how the manager reports timestamps in logs and APIs.
+        """,
+        examples=["UTC"],
+    )
+
+
+class APIConfig(BaseModel):
+    allow_origins: str = Field(
+        default="*",
+        description="""
+        CORS allow-origins setting.
+        Use '*' to allow all origins, or specify comma-separated domain patterns.
+        Important for browser-based clients connecting to the API.
+        """,
+        examples=["*", "https://example.com"],
+    )
+    allow_graphql_schema_introspection: bool = Field(
+        default=False,
+        description="""
+        Whether to allow GraphQL schema introspection.
+        Useful for development and debugging, but should be disabled in production.
+        When disabled, GraphQL tools like GraphiQL won't be able to explore the schema.
+        """,
+        examples=[True, False],
+    )
+    allow_openapi_schema_introspection: bool = Field(
+        default=False,
+        description="""
+        Whether to allow OpenAPI schema introspection.
+        Useful for development and debugging, but should be disabled in production.
+        When disabled, Swagger UI and similar tools won't work.
+        """,
+        examples=[True, False],
+    )
+    max_gql_query_depth: Optional[int] = Field(
+        default=None,
+        ge=1,
+        description="""
+        Maximum depth of GraphQL queries allowed.
+        Limits the complexity of queries to prevent abuse.
+        Set to None to disable the limit.
+        """,
+        examples=[None, 10, 15],
```

Comment:
How does `pydantic.Field` convert null value to `None`? Users should be able to set null values through config explicitly